src/Parser.php

```diff
@@ -158,14 +168,70 @@ private function findLibrary(): string
             '/usr/lib/',
         ];
 
+        // First, try to find platform-specific library
         foreach ($possiblePaths as $path) {
             $filePath = $path . $libraryName;
             if (file_exists($filePath)) {
                 return $filePath;
             }
         }
 
-        throw new ParserException('libpubliccode-parser.so not found');
+        // Fall back to legacy library name (for backward compatibility)
+        foreach ($possiblePaths as $path) {
+            $filePath = $path . $legacyLibraryName;
+            if (file_exists($filePath)) {
+                return $filePath;
+            }
+        }
```

Comment:
I think we should drop this logic. If we get here we'll try to load a x86_64 so on an unknown system & architecture, which should fail anyway.